data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -128,8 +131,17 @@ DataReaderRequest constructReaderRequest(final ReadRequest request) {
 
                 LOGGER.debug("Output stream closed, {} processed and {} returned, auditing success with audit service", recordsProcessed.get(), recordsReturned.get());
                 auditRequestComplete(readerRequest, recordsProcessed, recordsReturned);
-            } catch (IOException | NoCapacityException ex) {
+            } catch (IOException | RuntimeException ex) {
+                LOGGER.error("Exception thrown while reading data, auditing exception", ex);
```

Comment:
If you are throwing the exception then you dont need to log it here (SQ will complain about both logging and re-throwing. Suggest just audit logging here and then throw error to be acted on elsewhere. Or just change the logging line to log the exception message rather than the full exception